api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -139,10 +139,6 @@ const (
 type AzureManagedControlPlaneSpec struct {
 	AzureManagedControlPlaneClassSpec `json:",inline"`
 
-	// ResourceGroupName is the name of the Azure resource group for this AKS Cluster.
-	// Immutable.
-	ResourceGroupName string `json:"resourceGroupName"`
```

Comment:
Is this not API-breaking?

Reply:
It's similar to #4725 where it technically breaks the Go API by moving the field to the template type, but it wont break any user-facing scenarios. Nothing is removed from the CRD itself https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4723/files#diff-bab27b20a078754613e88030b4d2e38ba01555267143727b7f6718c10018cecd